--- /jgrapht-core/src/main/java/org/jgrapht/generate/WattsStrogatzGraphGenerator.java
+++ /jgrapht-core/src/main/java/org/jgrapht/generate/WattsStrogatzGraphGenerator.java
@@ -166,7 +166,7 @@ public class WattsStrogatzGraphGenerator<V, E>
 
         for (int i = 0; i < n; i++) {
             V vi = ring.get(i);
-            List<E> viAdj = adj.get(i);
+            List<E> viAdj = adj.get(vi);
 
             for (int j = 1; j <= k / 2; j++) {
                 viAdj.add(target.addEdge(vi, ring.get((i + j) % n)));
@@ -176,10 +176,9 @@ public class WattsStrogatzGraphGenerator<V, E>
         // re-wire edges
         for (int r = 0; r < k / 2; r++) {
             for (int i = 0; i < n; i++) {
-                V v = ring.get(i);
-                E e = adj.get(i).get(r);
-
                 if (rng.nextDouble() < p) {
+                    V v = ring.get(i);
+                    E e = adj.get(v).get(r);
                     V other = ring.get(rng.nextInt(n));
                     if (!other.equals(v) && !target.containsEdge(v, other)) {
                         if (!addInsteadOfRewire) {
