--- /spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java
+++ /spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java
@@ -178,7 +178,8 @@ public class SpannerTemplate implements SpannerOperations {
 
 	@Override
 	public <T> Page<T> queryAll(Class<T> entityClass, Pageable pageable) {
-		return queryAll(entityClass, pageable, null);
+		return queryAll(entityClass, pageable, new SpannerQueryOptions()
+				.setOffset(pageable.getOffset()).setLimit(pageable.getPageSize()));
 	}
 
 	@Override
