--- /src/main/java/edu/harvard/h2ms/domain/core/User.java
+++ /src/main/java/edu/harvard/h2ms/domain/core/User.java
@@ -36,7 +36,7 @@ public class User implements UserDetails {
   @Column(name = "ID")
   private Long id;
 
-  @NotNull @Column @JsonIgnore private String firstName;
+  @NotNull @Column private String firstName;
 
   @Column private String middleName;
 
@@ -50,7 +50,7 @@ public class User implements UserDetails {
 
   @NotNull @Column private String type;
 
-  @NotNull @Column private String password;
+  @NotNull @JsonIgnore @Column private String password;
 
   @ManyToMany
   @JoinTable(
