--- /src/main/java/org/springframework/data/jpa/util/JpaMetamodel.java
+++ /src/main/java/org/springframework/data/jpa/util/JpaMetamodel.java
@@ -76,7 +76,7 @@ public class JpaMetamodel {
 	public boolean isSingleIdAttribute(Class<?> entity, String name, Class<?> attributeType) {
 
 		return metamodel.getEntities().stream() //
-				.filter(it -> it.getJavaType().equals(entity)) //
+				.filter(it -> entity.equals(it.getJavaType())) //
 				.findFirst() //
 				.flatMap(it -> getSingularIdAttribute(it)) //
 				.filter(it -> it.getJavaType().equals(attributeType)) //
